src/main/java/com/github/skjolberg/packing/impl/PermutationRotationIterator.java

```diff
@@ -101,15 +103,15 @@ public PermutationRotationIterator(Dimension bound, PermutationRotation[] uncons
 		List<PermutationRotation> matrix = new ArrayList<>(unconstrained.length);
 		for(int i = 0; i < unconstrained.length; i++) {
 			List<Box> result = new ArrayList<>();
-			
+
 			Box[] boxes = unconstrained[i].getBoxes();
-			for(int k = 0; k < boxes.length; k++) {
-				if(boxes[k] != null && boxes[k].fitsInside3D(bound)) {
-					result.add(boxes[k]);
+			for (final Box boxe : boxes) {
```

Comment:
typo?

Reply:
yep, actually this is code generated by the IntelliJ IDE, I didn't check it knew the singular of boxes ;)

Reply:
fixed in 8a39f55